exercises/src/main/scala/fpinscala/parsing/JSON.scala

```diff
@@ -0,0 +1,63 @@
+package fpinscala.parsing
+
+trait JSON
+object JSON {
+  case object JNull extends JSON
+  case class JNumber(get: Double) extends JSON
+  case class JString(get: String) extends JSON
+  case class JBool(get: Boolean) extends JSON
+  case class JArray(get: IndexedSeq[JSON]) extends JSON
+  case class JObject(get: Map[String, JSON]) extends JSON
+
+  def parser[Parser[+_]](P: Parsers[Parser]): Parser[JSON] = {
+    import P.{string => _, _}
+    implicit val token = P.token _
+
+    val string: Parser[String] = {
+      val quote   = char('"')
+      val hex     = regex("[0-9a-fA-F]".r)
+      val escChar = char('"') | char('\\') | char('/') | char('b') |
+                    char('t') | char('n') | char('r') | char('t') |
+	            listOfN(4, hex)
+      val charEsc = char('\\') ~ escChar
+      val charLit = regex("[^\"\\\\]".r)
+      val letters  = (charLit | charEsc).many.slice
+
+      (quote ~ letters ~ quote) map (_._1._2)
```

Comment:
Agreed, using `*>` and `<*` is much better. I didn't have them initially, but added them later after inspecting the provided solution. Unfortunately, I overlooked this place when introducing it.

I don't "decode" string content, and this would be an obvious next step for improving the parser.